app/services/target_service/compatible_target_finder.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module TargetService
+  class CompatibleTargetFinder
+    def initialize(target)
+      @target = target
+      @user = target.user
+      @topic = target.topic
+    end
+
+    def call
+      find_compatible_targets
+    end
+
+    private
+
+    def match_targets
+      @match_targets ||= Target.where.not(user: @user).where(topic: @topic)
+    end
+
+    def find_compatible_targets
+      match_targets.each do |match_target|
+        distance = Geocoder::Calculations.distance_between([@target.latitude, @target.longitude],
+                                                           [match_target.latitude, match_target.longitude])
+        if distance <= @target.radius + match_target.radius
+          TargetNotification.with(user: @user).deliver_later(match_target.user)
```

Comment:
What happens after we create this Notification? Does the user receive an email or how is he being notified?

Reply:
It creates a new record in the notification model. The user can access to these notifications through the created relation. 
With this [gem](https://github.com/excid3/noticed), there are multiple ways to notice the user (database, email, slack,..). I chose database.

Reply:
Agree with @JulianPasquale, at least one method should be implemented and tested (email, push notification, etc). Not a blocker.

Reply:
Yeah, we should at least include and test one method. Right now notifications are being stored, but the user is not "being notified" since there's no way they can actually see that information.

We should send an email, or at least include a `/notifications` endpoint to get notifications. Otherwise we can't say we're "sending notifications"

Reply:
In order to notify the users when the targets match, an email will be sent. This is tested in the request spec `spec/requests/api/v1/targets/send_notification_spec.rb` 
I will add an spec for the CompatibleTargetFinder service validating all possible cases in a new PR.
@JulianPasquale @sebastiancaraballo